### PR TITLE
Add commands to induce contrived fatal scenarios

### PR DESCRIPTION
**Pull request by kdubau**
DEBUG builds only, useful for debugging Mono/MERP behavior

### CODE REVIEW

File: main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Commands.cs [hunk 1]
```diff
@@ -108,5 +112,40 @@ protected override async void Run ()
 			log.WriteLine ("=========================================================================================");
 		}
 	}
-}
 
+#if DEBUG
+	public class InduceManagedCrash : CommandHandler
```
Comment (Therzok) [inline main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Commands.cs]:
```suggestion
	class InduceManagedCrashHandler : CommandHandler
```

File: main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Commands.cs [hunk 2]
```diff
@@ -108,5 +112,40 @@ protected override async void Run ()
...
+	{
+		protected override void Run ()
+		{
+			Task.Run (() => {
+				new NSObject ().BeginInvokeOnMainThread (() => throw new Exception ("Diagnostics: Managed crash"));
+			});
+		}
+	}
+
+	public class InduceNativeCrash : CommandHandler
```
Comment (Therzok) [inline main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Commands.cs]:
```suggestion
	class InduceNativeCrashHandler : CommandHandler
```

File: main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Commands.cs [hunk 3]
```diff
@@ -108,5 +112,40 @@ protected override async void Run ()
...
+			Task.Run (() => {
+				using var x = new NSException ("Native crash", "Diagnostics", null);
+				var selector = ObjCRuntime.Selector.GetHandle ("raise");
+
+				void_objc_msgSend (x.Handle, selector);
+			});
+		}
+	}
+
+	public class InduceHang : CommandHandler
```
Comment (Therzok) [inline main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Commands.cs]:
```suggestion
	class InduceHangHandler : CommandHandler
```

File: main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Properties/ReliabilityDiagnostics.addin.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+﻿<?xml version="1.0" encoding="UTF-8"?>
+<ExtensionModel>
+
+	<Extension path="/MonoDevelop/Ide/Commands">
+
+		<Command id="PerformanceDiagnosticsAddIn.InduceManagedCrash"
+			defaultHandler="PerformanceDiagnosticsAddIn.InduceManagedCrash"
+			_label="Induce Managed Crash" />
+		<Command id="PerformanceDiagnosticsAddIn.InduceNativeCrash"
+			defaultHandler="PerformanceDiagnosticsAddIn.InduceNativeCrash"
```
Comment (Therzok) [inline main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Properties/ReliabilityDiagnostics.addin.xml]:
```suggestion
			defaultHandler="PerformanceDiagnosticsAddIn.InduceNativeCrashHandler"
```

File: main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Properties/ReliabilityDiagnostics.addin.xml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+			_label="Induce Native Crash" />
+		<Command id="PerformanceDiagnosticsAddIn.InduceHang"
+			defaultHandler="PerformanceDiagnosticsAddIn.InduceHang"
```
Comment (Therzok) [inline main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Properties/ReliabilityDiagnostics.addin.xml]:
```suggestion
			defaultHandler="PerformanceDiagnosticsAddIn.InduceHangHandler"
```

File: main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Commands.cs [hunk 6]
```diff
@@ -108,5 +112,40 @@ protected override async void Run ()
...
+	{
+		protected override void Run ()
+		{
+			Thread.Sleep (TimeSpan.FromMinutes(2));
+		}
+	}
+}
+#endif
```
Comment (Therzok) [inline main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Commands.cs]:
```suggestion
}
```

File: main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Commands.cs [hunk 7]
```diff
@@ -108,5 +112,40 @@ protected override async void Run ()
...
+	}
+
+	public class InduceHang : CommandHandler
+	{
+		protected override void Run ()
+		{
+			Thread.Sleep (TimeSpan.FromMinutes(2));
+		}
+	}
+}
```
Comment (Therzok) [inline main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Commands.cs]:
```suggestion
#endif
```

File: main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Properties/ReliabilityDiagnostics.addin.xml [hunk 8]
```diff
@@ -0,0 +1,26 @@
+﻿<?xml version="1.0" encoding="UTF-8"?>
+<ExtensionModel>
+
+	<Extension path="/MonoDevelop/Ide/Commands">
+
+		<Command id="PerformanceDiagnosticsAddIn.InduceManagedCrash"
+			defaultHandler="PerformanceDiagnosticsAddIn.InduceManagedCrash"
```
Comment (kdubau) [inline main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Properties/ReliabilityDiagnostics.addin.xml]: Missed an update here

Comment (Therzok) [reply main/src/addins/PerformanceDiagnostics/PerformanceDiagnostics/Properties/ReliabilityDiagnostics.addin.xml]:
```suggestion
			defaultHandler="PerformanceDiagnosticsAddIn.InduceManagedCrashHandler"
```

**Comment by kdubau** [comment]
You guys are quick :)

**Pull request merged**